### PR TITLE
fix master locked bug and add load bucket error check

### PR DESCRIPTION
**Pull request by KKRainbow**

### CODE REVIEW

File: executor/worker.go [hunk 1]
```diff
@@ -111,14 +122,29 @@ func (w *Worker) Run() {
...
-		})
-		// backoff
-		if err != nil {
-			time.Sleep(time.Duration(rand.IntnRange(1, 3)) * time.Second)
+		timerStopped <- true
+
+		// synchronize with heartbeat goroutine
+		<- lastHeartbeatSent
+
+		for {
```
Comment (Lcch) [inline executor/worker.go]: Why do we need to for-loop on this part? Is it possible that server detect time-out for this worker by accident, however, this worker will keep reporting job result due to ret.Retcode != 0

File: master/master.go [hunk 2]
```diff
@@ -101,6 +102,21 @@ func (m *Master) JobFinished(job *jobgraph.Job) {
 	m.waitFinish.Done()
 }
 
+func (m *Master) resetWorker(workerID int64) {
+	m.Lock()
+	defer m.Unlock()
+Loop:
```
Comment (Lcch) [inline master/master.go]: it will never end?

File: executor/worker.go (hunk 1 above)
Comment (KKRainbow) [reply executor/worker.go]: ReportJob always return ret.Retcode=0
error != nil when network error happens

File: master/master.go (hunk 2 above)
Comment (KKRainbow) [reply master/master.go]: It will when execute `break Loop`

**Pull request merged**